packages/frontend-2/lib/viewer/composables/ui.ts

```diff
@@ -339,3 +339,30 @@ export function useThreadUtilities() {
 
   return { closeAllThreads, open, isOpenThread }
 }
+
+export function useMeasurementUtilities() {
+  const { instance } = useInjectedViewer()
```

Comment:
These utilities are interacting with the viewer directly, foregoing the viewer state. This is different from all of the other integrations between FE2 and Viewer are implemented and cause the following problems:

1. Interacting with the viewer from FE2 requires you to imperatively trigger functions onto the viewer instance. We want the integration to be declarative so that it's idiomatic to Vue apps. Instead of imperatively triggering measurements on/off, it would be more idiomatic to have a `areMeasurementsOn: boolean` reactive ref that you can read and write to. In this case you've encapsulated all of the imperative logic within your utilities, which is good, cause then devs don't have to do that manually, but all of this still exists outside of the declarative/reactive viewer state and thus can't be watched, mutated idempotently etc.
2. Since none of this is a part of the viewer state, we can't read the current state (are measurements currently enabled?) for UI purposes or for serialization purposes. So we can't accurately show that measurements are enabled in the UI or serialize this data for comments/follow mode (e.g. how we serialize and save other viewer settings with comments, so that when u open the comment u get the same viewer settings).

So, ideally, this should be implemented like the rest of the FE2<->Viewer integration:

1. Add new properties to the viewer state relevant to this integration, probably under  `state.ui`. Maybe something like: `{ui: {measurements: {areEnabled: Ref<boolean>, options: Ref<undefined | MeasurementOptions>}}}`. This will allow you to declaratively read the current state and also write new measurement settings.
2. In `postSetup.ts` actually create a composable for your integration and invoke it in `useViewerPostSetup()`. You can look at how other integrations are set up, but you would probably define `watch`ers on the new state properties, that would send the new state values to the viewer instance (instance.enableMeasurements/instance.setMeasurementOptions), only if they've changed.
4. You can leave a `useMeasurementUtilities()` that you created in this file, but it should only change those new state refs inside of them, instead of interacting with the viewer directly.

So essentially: 1. Only create the refs themselves in the viewer state, adjust TS types etc. 2. Actually implement the logic for reading/writing from/to these refs in postSetup 3. If needed, create utilities in `viewer/composables/` that simplify working with the viewer state.

Reply:
Once we have this data in the viewer state, we can also adjust the serialization logic so that these settings are stored with comments or sent to followers in spotlight mode, but I'm not sure if we need that currently